### ISSUE TITLE
nll_loss gives result when both input and target are 1D tensor

### ISSUE THREAD

**Issue by Bin1024** [pytorch/pytorch#157420]
### 🐛 Describe the bug

According to the [official document](https://docs.pytorch.org/docs/2.7/generated/torch.nn.functional.nll_loss.html#torch.nn.functional.nll_loss), nll_loss can compute case `input.shape=(N, C)` and `target.shape=(N,)`, where `N` is batch number and `C` is numbert of classes, in this case `input.dim=2` and `target.dim=1`.
When N=1, it can be simplified to `input.shape=(C), target.shape=()`, in this case `input.dim=1, target.dim=0`. 
So what is the meaning of case `input.dim=1, target.dim=1`? nll_loss gives me the result of this case, the sample code are as following:
```python
import torch
import torch.nn.functional as F

input_tensor = torch.randn((10,)).cuda()
target_tensor = torch.randint(0, 5, (10,)).cuda()

loss = F.nll_loss(input_tensor, target_tensor)

print("Final Loss:", loss.item())
```

### Versions

torch: 2.6.0.dev20241112+cu121

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

**Comment by mansiag05** [comment]
I would like to pick up this issue. Will raise a PR for the same.
